src/workload.rs

```diff
@@ -639,10 +620,14 @@ impl WorkloadStore {
                 }
             }
         }
-        if Some(&w.node) == self.local_node.as_ref()
-            || (Some(&w.name) == self.local_pod.as_ref()
-                && Some(&w.namespace) == self.local_namespace.as_ref())
-        {
+        let should_fetch: bool = match self.proxy_mode {
+            ProxyMode::Sidecar => {
```

Comment:
For sidecar we can just always pre-getch our own identity, even before we get xds response. This is simpler and faster

Reply:
Yes, I think so, let me optimize it in the next PR, ok?

Reply:
I am a bit concerned with doing any pre-fetching in sidecar mode. can we change this to never do it for sidecars mode in the short term then?

Reply:
> can we change this to never do it for sidecars mode in the short term then?

@howardjohn  OK, I changed to not pre-fetch cert in sidecar mode, and tested in the real environment, for the first request, there is not much performance impact.